skopt/searchcv.py

```diff
@@ -0,0 +1,425 @@
+from collections import defaultdict
+
+import numpy as np
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Parallel, delayed, cpu_count
+import sklearn.model_selection._search as sk_model_sel
+
+from skopt import Optimizer
+from skopt.utils import point_asdict, dimensions_aslist
+
+class BayesSearchCV(sk_model_sel.BaseSearchCV):
+    """Bayesian optimization over hyper parameters.
+
+    BayesSearchCV implements a "fit" and a "score" method.
+    It also implements "predict", "predict_proba", "decision_function",
+    "transform" and "inverse_transform" if they are implemented in the
+    estimator used.
+
+    The parameters of the estimator used to apply these methods are optimized
+    by cross-validated search over parameter settings.
+
+    In contrast to GridSearchCV, not all parameter values are tried out, but
+    rather a fixed number of parameter settings is sampled from the specified
+    distributions. The number of parameter settings that are tried is
+    given by n_iter.
+
+    Parameters are presented as a list of skopt.space.Dimension objects.
+
+    Parameters
+    ----------
+    estimator : estimator object.
+        A object of that type is instantiated for each search point.
+        This object is assumed to implement the scikit-learn estimator api.
+        Either estimator needs to provide a ``score`` function,
+        or ``scoring`` must be passed.
+
+    search_spaces : list of dict or tuple
```

Comment:
While this is nice, the 90% use case is to search over a single space. I think we should support directly providing a dict. This would also better match with the API of `GridSearchCV`.